src/HdpExecutionStore.sol

```diff
@@ -122,15 +125,19 @@ contract HdpExecutionStore is Initializable, UUPSUpgradeable, OwnableUpgradeable
         bytes32 datalakeCommitment = blockSampledDatalake.commit();
         bytes32 taskCommitment = computationalTask.commit(datalakeCommitment);
 
-        // Ensure task is not already scheduled
-        if (cachedTasksResult[taskCommitment].status != TaskStatus.NONE) {
-            revert DoubleRegistration();
-        }
+        if (cachedTasksResult[taskCommitment].status == TaskStatus.FINALIZED) {
+            emit TaskAlreadyStored(taskCommitment);
```

Comment:
Shouldn't you return to end the execution?

Reply:
But if requested task does not exist, we cannot return the value no? Is it possible to choose the return type per branch in solidity?

Reply:
Smth like if return type can be Option type this can be solved -- not sure if it doable in sol

Reply:
nvm, all good. Approving